misc/runtime_upgrade/main.py

```
@@ -0,0 +1,93 @@
+    parser.add_argument('--node-url', help='URL of the node to connect to', dest='node_url', default='ws://127.0.0.1:9944', required=False)
+    parser.add_argument('--wasm-file-path', help='Path to Compressed Wasm File', dest='wasm_file_path', required=True)
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--uri', help='URI of the keypair to use', dest='uri_keypair', type=str)
+    group.add_argument('--seed', help='Seed of the keypair to use', dest='seed', type=str)
+    group.add_argument('--mnemonic', help='Seed phrase of the keypair to use', dest='mnemonic', type=str)
+    args = parser.parse_args()
+    return args
+
+def get_keypair_using_args(args):
+    if args.uri_keypair:
+        return Keypair.create_from_uri(args.uri_keypair, ss58_format=ss58_format)
+    elif args.seed:
+        return Keypair.create_from_seed(args.seed, ss58_format=ss58_format)
+    elif args.mnemonic:
+        return Keypair.create_from_mnemonic(args.mnemonic, ss58_format=ss58_format)
+    else:
+        raise Exception("No keypair provided")
+    
+
+def send_extrinsic(substrate_provider: SubstrateInterface, keypair: Keypair, call: GenericCall):
+    extrinsic = substrate_provider.create_signed_extrinsic(call=call, keypair=keypair)
+    try:
+        receipt = substrate_provider.submit_extrinsic(extrinsic, wait_for_inclusion=True)
+        log_receipt(receipt)
+    except SubstrateRequestException as e:
+        print(f'Error in send_extrinsic: {e}')
+        raise e
+
+def log_receipt(receipt):
+    print('Extrinsic "{}" included in block "{}"'.format(
+        receipt.extrinsic_hash, receipt.block_hash
+    ))
+
+    if receipt.is_success:
+        print('✅ Success, triggered events:')
+        for event in receipt.triggered_events:
+            print(f'* {event.value}')
+    else:
+        print('⚠️ Extrinsic Failed: ', receipt.error_message)
```

I think it is better to raise an Exception to return 1 from the script.